packages/admin/cms-admin/src/generator/future/generateForm.ts

```diff
@@ -62,10 +72,10 @@ export function generateForm(
             update${gqlType}(id: $id, input: $input, lastUpdatedAt: $lastUpdatedAt) {
                 id
                 updatedAt
-                ...${fragmentName}
+                ...${updateMutationFragmentName}
```

Comment:
ok, you are adding a second fragment that you use for the update mutation.

That fragment is used for reading though! It rquests the fields that get updated using the mutation response.

I see no win in removing the readOnly fields:

- it's a neglectable performance improvement
- it complicates the generated code (more fragments)
- it doesn't update the readOnly fields on mutate, although they _might_ be modified (using some server side business code) on update (example: updatedAt)

And this fragment does _not_ prevent the readOnly fields getting sent as input
- imho overwriting an readOnly field is not good practice
- that could cause permission problems, if not all users are allowed to modify all fields

What could be done about it?
- excluding those fields when building input
- or make the readOnly fields not part of the final-form (using a second fragment containing readOnly fields) and output the readOnly fields not as `<Field`

Reply:
Readonly fields shouldn't be sent as part of the input, we need to change that.

Also, I wouldn't create a second fragment that omits the readonly field. There could even be cases where this is wrong. For instance, an update may cause a dependent readonly "field" to change (e.g., changing to a different organization could change the required license), and this update wouldn't be reflected in the form after saving, because the cache wouldn't be updated.

Reply:
> That fragment is used for reading though!

Ok, I misunderstood how fragments/mutations work 😅 
Will remove that update-fragment stuff. 

> excluding those fields when building input

I suppose this is the best way to do this then, as this does not require changes to how the form works. 
Should I do it this way?

Reply:
@nsams @johnnyomair I removed the unnecessary update-fragment. 

Also the read-only fields are not no longer part of the input to the update-mutation. 
I'm not sure however, if this should also be the case for the create-mutation, as a required field that is read-only will cause a typescript error because it must not be undefined. 

Should we:
- Just send the full input, including the read-only fields during create?
- Exclude them and ignore the fact that required read-only fields will cause type errors? 
- Exclude them, but throw an error when trying to generate a required read-only field?

I'm not sure if there is a use-case for required read-only fields, except maybe required fields, that can only be changed with certain user-permissions 🤔

Reply:
I would not care too much about that for now

read-only + create doesn't make a lot of sense. And a read-only & required field + create even less.